sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -88,3 +92,36 @@ def test_extract_intro():
             result,
             'This is the description of the example which goes on and on')
         assert_false('second paragraph' in result)
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
+    examples_dir = tempfile.mkdtemp()
+    gallery_dir = tempfile.mkdtemp()
+
+    gallery_conf = {
+        'filename_pattern': '/plot_0',
```

Comment:
use os.path.sep
Travis will not complain but it'll fail for people testing in windows